addons/website/static/src/snippets/s_map/options.js

```diff
@@ -0,0 +1,93 @@
+odoo.define('options.s_map_options', function (require) {
+'use strict';
+
+const {_t} = require('web.core');
+const options = require('web_editor.snippets.options');
+
+options.registry.Map = options.Class.extend({
+    /**
+     *
+     * @override
+     */
+    onBuilt: function () {
+        this.updateUI();
+    },
+
+    /**
+     * @override
+     */
+    updateUI() {
+        const dataset = this.$target[0].dataset;
+        const embedded = this.$target.find('.o_embedded');
+        const info = this.$target.find('.missing_option_warning');
+        if (dataset.mapAddress) {
+            const url = 'https://maps.google.com/maps?q=' + encodeURIComponent(dataset.mapAddress)
+                + '&t=' + encodeURIComponent(dataset.mapType)
+                + '&z=' + encodeURIComponent(dataset.mapZoom)
+                + '&ie=UTF8&iwloc=&output=embed';
+            if (url !== embedded.attr('src')) {
+                embedded.attr('src', url);
+            }
+            embedded.removeClass('d-none');
+            info.addClass('d-none');
+        } else {
+            embedded.attr('src', 'about:blank');
+            embedded.addClass('d-none');
+            info.removeClass('d-none');
+        }
+        if (! (this.$target.hasClass('o_half_screen_height') || this.$target.hasClass('o_full_screen_height'))) {
+            this.$target.css('min-height', this.$target[0].dataset.mapMinHeight);
+        }
+        return this._super.apply(this, arguments).then(() => {
+            this.updateUIVisibility();
+        });
+    },
+
+    //--------------------------------------------------------------------------
+    // Options
+    //--------------------------------------------------------------------------
+
+    /**
+     * @see this.selectClass for parameters
+     */
+    async showDescription(previewMode, widgetValue, params) {
+        const descriptionEl = this.$target[0].querySelector('.description');
+        if (widgetValue && !descriptionEl) {
+            this.$target.append($(`
+                <div class="description">
+                    <font>${_t('Visit us:')}</font>
+                    <span>${_t('Our office is open Monday – Friday 8:30 a.m. – 4:00 p.m.')}</span>
+                </div>`)
+            );
+        } else if (!widgetValue && descriptionEl) {
+            descriptionEl.remove();
+        }
+    },
+    /**
+     * @see this.selectClass for parameters
+     */
+    async setHeight(previewMode, widgetValue, params) {
+        if (widgetValue) {
+            this.$target[0].dataset.mapMinHeight = widgetValue;
+            this.$target.css('min-height', widgetValue);
+        }
+    },
```

Comment:
I don't think that option should have been added at all (@sbuhl ? Height for map snippet only ? :thinking:), we have the overlay to control that... but if we really want to keep it:
- Get rid of that `dataset.mapMinHeight`, that is adding duplicated info on the DOM for no reason at all. Basically, that whole `setHeight` method is useless as you can use the generic `selectStyle` method to do this and would allow to remove your whole JS.
- Enable preview mode (would be automatic by using `selectStyle` I guess

Reply:
Indeed the pixel height option seems useless.
The only situation I see where this option helps is for example when putting the map inside a column layout, somehow using the overlay has no effect on the height - but then it would indeed be cleaner to remove the pixel height setting on this snippet and find a way to make the overlay resize work in those situations.

Reply:
Pixel height was not useless when merged, so please double check before to remove it.